mlrun/api/api/utils.py

```diff
@@ -59,19 +60,29 @@ def get_obj_path(schema, path, user=""):
         data_volume_prefix = config.httpdb.data_volume
         if data_volume_prefix.endswith("/"):
             data_volume_prefix = data_volume_prefix[:-1]
-        if config.httpdb.real_path:
+        if real_path:
             path_from_volume = path[len(data_volume_prefix) :]
             if path_from_volume.startswith("/"):
                 path_from_volume = path_from_volume[1:]
-            path = str(Path(config.httpdb.real_path) / Path(path_from_volume))
+            path = str(Path(real_path) / Path(path_from_volume))
     if schema:
         schema_prefix = schema + "://"
         if not path.startswith(schema_prefix):
             path = f"{schema_prefix}{path}"
-    if not path.startswith("v3io://") and (
-        not config.httpdb.real_path
-        or (config.httpdb.real_path and not path.startswith(config.httpdb.real_path))
-    ):
+
+    # Check if path is allowed - v3io:// is always allowed, and also the real_path parameter if specified.
+    # We never allow local files in the allowed paths list. Allowed paths must contain a schema (://)
+    allowed_file_paths = config.httpdb.allowed_file_paths
+    allowed_paths_list = (
+        [path.strip() for path in allowed_file_paths.split(",") if "://" in path]
+        if allowed_file_paths
+        else []
+    )
```

Comment:
```suggestion
    allowed_file_paths = config.httpdb.allowed_file_paths or ""
    allowed_paths_list = [path.strip() for path in allowed_file_paths.split(",") if "://" in path]
```

Reply:
Just a suggestion to make the code more concise

Reply:
Issue with this is that `split` *with delimiter* on an empty string will return a non-empty array (`['']`). Then checking `startswith` on the empty string will not give us correct results.
It is an annoying issue with the fact that `split` implements two different mechanisms when a delimiter is provided and when it is not - [see here](https://stackoverflow.com/questions/16645083/when-splitting-an-empty-string-in-python-why-does-split-return-an-empty-list#:~:text=In%20the%20case%20of%20splitting,produce%20the%20first%20empty%20field.)

Reply:
After discussing this, made the change.